src/Tickets/Order_Modifiers/Checkout/Abstract_Fees.php

```diff
@@ -0,0 +1,319 @@
+<?php
+/**
+ * Fees Class for handling fee logic in the checkout process.
+ *
+ * This class is responsible for managing fees, calculating them, and appending them
+ * to the cart. It integrates with various filters and hooks during the checkout process.
+ *
+ * @since TBD
+ * @package TEC\Tickets\Order_Modifiers\Checkout
+ */
+
+namespace TEC\Tickets\Order_Modifiers\Checkout;
+
+use TEC\Tickets\Commerce\Gateways\Contracts\Gateway_Interface;
+use TEC\Tickets\Commerce\Utils\Value;
+use TEC\Tickets\Order_Modifiers\Controller;
+use TEC\Tickets\Order_Modifiers\Modifiers\Modifier_Manager;
+use TEC\Tickets\Order_Modifiers\Modifiers\Modifier_Strategy_Interface;
+use TEC\Tickets\Order_Modifiers\Repositories\Order_Modifier_Relationship;
+use TEC\Tickets\Order_Modifiers\Repositories\Order_Modifiers;
+
+/**
+ * Class Fees
+ *
+ * Handles fees logic in the checkout process.
+ *
+ * @since TBD
+ */
+abstract class Abstract_Fees {
+
+	/**
+	 * The modifier type used for fees.
+	 *
+	 * @since TBD
+	 * @var string
+	 */
+	protected string $modifier_type = 'fee';
+
+	/**
+	 * The modifier strategy for applying fees.
+	 *
+	 * @since TBD
+	 * @var Modifier_Strategy_Interface
+	 */
+	protected Modifier_Strategy_Interface $modifier_strategy;
+
+	/**
+	 * Manager for handling modifier calculations and logic.
+	 *
+	 * @since TBD
+	 * @var Modifier_Manager
+	 */
+	protected Modifier_Manager $manager;
+
+	/**
+	 * Repository for accessing order modifiers.
+	 *
+	 * @since TBD
+	 * @var Order_Modifiers
+	 */
+	protected Order_Modifiers $order_modifiers_repository;
+
+	/**
+	 * Repository for accessing and managing relationships between order modifiers.
+	 *
+	 * @since TBD
+	 * @var Order_Modifier_Relationship
+	 */
+	protected Order_Modifier_Relationship $order_modifiers_relationship_repository;
+
+	/**
+	 * Subtotal value used in fee calculations.
+	 *
+	 * This represents the total amount used as a basis for calculating applicable fees.
+	 *
+	 * @since TBD
+	 * @var float
+	 */
+	protected float $subtotal = 0.0;
+
+	/**
+	 * Tracks whether the fees have already been displayed during the checkout process.
+	 *
+	 * This static property ensures that the fees are only displayed once across multiple instances
+	 * of the class. If set to `true`, the hook for displaying fees will not be added again.
+	 * The default is `false`, indicating the fees have not yet been displayed.
+	 *
+	 * @since TBD
+	 * @var bool
+	 */
+	protected static bool $fees_displayed = false;
+
+	/**
+	 * Tracks whether the fees have already been appended to the cart.
+	 *
+	 * This static property ensures that the fees are only appended once during the
+	 * checkout process across multiple instances of the class. If set to `true`, the
+	 * method `append_fees_to_cart` will not add the fees again. The default is `false`,
+	 * indicating the fees have not yet been appended.
+	 *
+	 * @since TBD
+	 * @var bool
+	 */
+	protected static bool $fees_appended = false;
```

Comment:
I would rather see this stored in the cart structure, and not mix responsibilities, feels like it would be a pain to find this here after a couple of iterations of code.